src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Semantic/SemanticTokenTestBase.cs

```diff
@@ -69,45 +69,28 @@ protected void AssertSemanticTokensMatchesBaseline(IEnumerable<int>? actualSeman
         var baselineFileName = Path.ChangeExtension(fileName, ".semantic.txt");
         var actual = actualSemanticTokens?.ToArray();
 
+        var actualFileContents = GetFileRepresentationOfTokens(actual);
+
         BaselineTestCount++;
         if (GenerateBaselines)
         {
-            GenerateSemanticBaseline(actual, baselineFileName);
-        }
-
-        var semanticArray = GetBaselineTokens(baselineFileName);
-
-        if (semanticArray is null && actual is null)
-        {
-            return;
-        }
-        else if (semanticArray is null || actual is null)
-        {
-            Assert.False(true, $"Expected: {semanticArray}; Actual: {actual}");
+            GenerateSemanticBaseline(actualFileContents, baselineFileName);
         }
 
-        for (var i = 0; i < Math.Min(semanticArray!.Length, actual!.Length); i += 5)
-        {
-            var end = i + 5;
-            var actualTokens = actual[i..end];
-            var expectedTokens = semanticArray[i..end];
-            Assert.True(Enumerable.SequenceEqual(expectedTokens, actualTokens), $"Expected: {string.Join(',', expectedTokens)} Actual: {string.Join(',', actualTokens)} index: {i}");
-        }
+        var expectedFileContents = GetBaselineFileContents(baselineFileName);
 
-        Assert.True(semanticArray.Length == actual.Length, $"Expected length: {semanticArray.Length}, Actual length: {actual.Length}");
+        new XUnitVerifier().EqualOrDiff(expectedFileContents, actualFileContents);
```

Comment:
maybe changed later, but doesn't this collide with your other PR?

Reply:
Yeah, now that the other one has merged this won't compile any more.